### PR TITLE
Cardano wallet

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/hooks/index.ts [hunk 1]
```diff
@@ -14,20 +14,19 @@ import {
 } from 'react-query'
 
 import {isNightly} from '../legacy/config'
-import {ObjectValues} from '../legacy/flow'
```
Comment (stackchain) [inline src/hooks/index.ts:17]: 👀

File: src/yoroi-wallets/cardano/CardanoWallet.ts [hunk 2]
```diff
@@ -109,8 +102,8 @@ export type ByronWalletJSON = Omit<ShelleyWalletJSON, 'account'>
 
 export type WalletJSON = ShelleyWalletJSON | ByronWalletJSON
 
-export default ShelleyWallet
-export class ShelleyWallet implements WalletInterface {
+export default CardanoWallet
+export class CardanoWallet implements YoroiWallet {
```
Comment (stackchain) [inline src/yoroi-wallets/cardano/CardanoWallet.ts:106]: 🤌

File: src/yoroi-wallets/utils/amountUtils.ts [hunk 3]
```diff
@@ -54,3 +52,5 @@ export const editedFormatter = compose(
   formatMultiLangSeparator,
   stripInvalidCharacters,
 )
+
+const insertAt = (str: string, pos: number, text: string) => [str.slice(0, pos), text, str.slice(pos)].join('')
```
Comment (stackchain) [inline src/yoroi-wallets/utils/amountUtils.ts:56]: maybe `${str.slice(0, pos)}${text}${str.slice(pos)}` dunno

File: src/yoroi-wallets/walletManager/walletManager.ts [hunk 4]
```diff
@@ -2,19 +2,30 @@
...
 export class WalletClosed extends ExtendableError {}
 
+export type WalletMeta = {
+  id: string
+  name: string
+  networkId: NetworkId
+  walletImplementationId: WalletImplementationId
+  isHW: boolean
+  isShelley?: boolean | null | undefined
+  // legacy jormungandr
```
Comment (stackchain) [inline src/yoroi-wallets/walletManager/walletManager.ts:24]:
```suggestion
  // isShelly is legacy jormungandr
  isShelley?: boolean | null | undefined

```

Comment (stackchain) [reply src/yoroi-wallets/walletManager/walletManager.ts:24]: dunno why is out of order

**Review by stackchain — APPROVED** [review]

File: src/legacy/format.ts [hunk 5]
```diff
@@ -8,125 +8,82 @@ import {defineMessages} from 'react-intl'
...
-  token: Token | DefaultAsset,
-  denomination: AssetDenomination,
-  intl: IntlShape,
-): string => getAssetDenomination(token, denomination) ?? intl.formatMessage(messages.unknownAssetName)
 export const normalizeTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): BigNumber => {
   const normalizationFactor = Math.pow(10, token.metadata.numberOfDecimals)
   return amount.dividedBy(normalizationFactor).decimalPlaces(token.metadata.numberOfDecimals)
 }
+
 export const formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string => {
```
Comment (stackchain) [inline src/legacy/format.ts:46]: Maybe overload it, to work closer with Quantities/Entries/Amounts?

formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string
formatTokenAmount = (amount: string, token: Token | DefaultAsset): string
```suggestion
export const formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string => {
```

**Review by stackchain — COMMENTED** [review]
⬆️

**Comment by wolverineks** [comment]
mostly file moves, and renames
legacy/format is the only file that needs a double check

**Pull request closed without merging**